Sapling comparison docs update

AngelEzquerra opened the pull request:
Update the comparison to sapling docs to take into account some of the features that have been added to jujutsu since they were last updated.

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

PhilipMetzger commented:
Can be a single commit and should have the title `docs:` per https://github.com/jj-vcs/jj/blob/main/docs/contributing.md.

AngelEzquerra commented:
> Can be a single commit and should have the title `docs:` per https://github.com/jj-vcs/jj/blob/main/docs/contributing.md.

I apologize. I had to use "bare" git to push to GitHub since jj push was not working for me. I did not realize that I was also pushing the empty head commit. I have removed the empty commit and updated the description as you requested.

PhilipMetzger reviewed (approved):
Thanks, lgtm.
